src/compiler/checker.ts

```diff
@@ -37767,7 +37767,7 @@ namespace ts {
                 return;
             }
             // Grammar checking
-            if (!checkGrammarDecoratorsAndModifiers(node) && hasEffectiveModifiers(node)) {
+            if (node.isExportEquals && !checkGrammarDecoratorsAndModifiers(node) && hasEffectiveModifiers(node)) {

```

Comment:
This seems unrelated/unnecessary; please revert if that’s the case.

Reply:
My mistake. With your help, this PR has gone through a total rewrite. At first I changed compiler(which is not a good direction), then i rewrite as you suggested. These code are old commit, and absolutely should be removed.